src/components/AddPeopleAndRoles/OrgPerson.vue

```diff
@@ -0,0 +1,533 @@
+      }
+    }
+  }
+
+  // Event Handlers
+  private updateMailingAddress (val): void {
+    this.inProgressMailingAddress = val
+  }
+
+  private updateDeliveryAddress (val): void {
+    this.inProgressDeliveryAddress = val
+  }
+
+  private updateMailingAddressValidity (val: boolean): void {
+    this.mailingAddressValid = val
+  }
+
+  private updateDeliveryAddressValidity (val: boolean): void {
+    this.deliveryAddressValid = val
+  }
+
+  private assignCompletingPartyRole (): void {
+    if (this.isCompletingParty && this.existingCompletingParty &&
+      this.orgPerson.id !== this.existingCompletingParty.id) {
+      this.confirmReassignPerson()
+    }
+  }
+
+  // Methods
+  private validateAddPersonOrgForm (): void {
+    if (this.isFormValid()) {
+      if (this.reassignCompletingParty) {
+        this.emitReassignCompletingPartyEvent()
+      }
+      const person: OrgPersonIF = this.addPerson()
+      this.emitPersonInfo(person)
+      this.resetAddPersonData(false)
+    }
+  }
+
+  private isFormValid () : boolean {
+    let isFormValid: boolean = this.addPersonOrgFormValid && this.mailingAddressValid
+    if (this.isDirector && !this.inheritMailingAddress) {
+      isFormValid = isFormValid && this.deliveryAddressValid
+    }
+    return isFormValid
+  }
+
+  private confirmReassignPerson () {
+    // open confirmation dialog and wait for response
+    this.$refs.reassignCPDialog.open(
+      'Change Completing Party?',
+      this.reassignPersonErrorMessage(),
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Change Completing Party',
+        no: null,
+        cancel: 'Cancel'
+      }
+    ).then(async (confirm) => {
+      if (confirm) {
+        this.reassignCompletingParty = true
+      }
+    }).catch(() => {
+      this.isCompletingParty = false
+    })
+  }
+
+  /**
+   * Local helper push the current director data into the list.
+   */
+  private addPerson (): OrgPersonIF {
+    let personToAdd: OrgPersonIF = { ...this.orgPerson }
+    if (this.activeIndex === -1) {
+      personToAdd.id = this.nextId
+    }
+    personToAdd.address = this.setPersonAddress()
+    personToAdd.roles = this.setPersonRoles()
+    return personToAdd
+  }
+
+  private setPersonAddress (): BaseAddressObjIF {
+    let personAddress: BaseAddressObjIF = {
+      mailingAddress: { ...this.inProgressMailingAddress }
+    }
+    if (this.isDirector) {
+      if (this.inheritMailingAddress) {
+        this.inProgressDeliveryAddress = this.inProgressMailingAddress
+      }
+      personAddress = { ...personAddress, deliveryAddress: { ...this.inProgressDeliveryAddress } }
+    }
+    return personAddress
+  }
+
+  private setPersonRoles (): string [] {
+    let roles: string[] = []
+    if (this.isCompletingParty) {
+      roles.push(Roles.COMPLETING_PARTY)
+    }
+    if (this.isDirector) {
+      roles.push(Roles.DIRECTOR)
+    }
+    if (this.isIncorporator) {
+      roles.push(Roles.INCORPORATOR)
```

Comment:
Per mockup, Incorporator should come before Director.

Reply:
fixed!